hazelcast/src/main/java/com/hazelcast/internal/util/phonehome/CacheInfoCollector.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hazelcast.internal.util.phonehome;
+
+import com.hazelcast.cache.impl.CacheService;
+import com.hazelcast.config.CacheSimpleConfig;
+import com.hazelcast.core.DistributedObject;
+import com.hazelcast.instance.impl.Node;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+class CacheInfoCollector implements MetricsCollector {
+
+    @Override
+    public Map<String, String> computeMetrics(Node hazelcastNode) {
+        Map<String, String> cacheInfo = new HashMap<>(1);
+
+        Collection<DistributedObject> distributedObjects = hazelcastNode.hazelcastInstance.getDistributedObjects();
+        long countCacheWithWANReplication = distributedObjects.stream()
+                .filter(distributedObject -> distributedObject.getServiceName().equals(CacheService.SERVICE_NAME))
+                .filter(distributedObject -> {
```

Comment:
Hello @chanmol1999 thanks for addressing my previous comment. Now if you look at the code, you can see that the lambda is still a little bit long, so we can think a bit about if we can break it up into more smaller strem operations: you can observe that
 * the filtering lambda uses only the config of the cache object
 * no subsequent stream operation uses the cache object, that the stream operates on

So given this, first you can use the `map()` operation to map the cache object to its config, then change the current (second) filter to perform the check on the config, which then will be possible in a one-liner lambda.

Please let me know on gitter if anyhting is unclear.